core/src/main/java/org/jruby/RubyArithmeticSequence.java

```diff
@@ -0,0 +1,565 @@
+        Ruby runtime = context.runtime;
+
+        IRubyObject v;
+
+        v = safeHash(context, excludeEnd);
+        long hash = hashStart(runtime, v.convertToInteger().getLongValue());
+
+        v = safeHash(context, begin);
+        hash = murmurCombine(hash, v.convertToInteger().getLongValue());
+
+        v = safeHash(context, end);
+        hash = murmurCombine(hash, v.convertToInteger().getLongValue());
+
+        v = safeHash(context, step);
+        hash = murmurCombine(hash, v.convertToInteger().getLongValue());
+        hash = hashEnd(hash);
+
+        return runtime.newFixnum(hash);
+    }
+
+    @Override
+    public final IRubyObject inspect() {
+        return inspect(getRuntime().getCurrentContext());
+    }
+
+    @JRubyMethod
+    public RubyString inspect(ThreadContext context) {
+        ByteList str = new ByteList();
+        boolean isRange = generatedBy instanceof RubyRange;
+        str.append('(');
+        if (isRange) {
+            str.append('(');
+        }
+        str.append(RubyObject.inspect(context, generatedBy).getByteList());
+        if (isRange) {
+            str.append(')');
+        }
+        str.append('.').append(method.getBytes());
+        if (args != null) {
+            int argc = args.length;
+
+            if (argc > 0) {
+                IRubyObject kwds = context.nil;
+
+                str.append('(');
+
+                if (args[argc - 1] instanceof RubyHash) {
+                    boolean allKey = true;
+                    IRubyObject [] keys = ((RubyHash)args[argc - 1]).keys().toJavaArray();
+                    for (IRubyObject key : keys) {
+                        if (!(key instanceof RubySymbol)) {
+                            allKey = false;
+                            break;
+                        }
+                    }
+                    if (allKey) {
+                        kwds = args[--argc];
+                    }
+                }
+
+                while (argc > 0) {
+                    str.append(RubyObject.inspect(context, args[args.length - argc]).getByteList());
+                    str.append(',').append(' ');
+                    --argc;
+                }
+                
+                if (!kwds.isNil()) {
+                    IRubyObject [] keys = ((RubyHash)kwds).keys().toJavaArray();
+                    for (IRubyObject key : keys) {
+                        IRubyObject value = ((RubyHash)kwds).fastARef(key);
+                        str.append(((RubySymbol)key).getBytes());
+                        str.append(':').append(' ');
+                        str.append(RubyObject.inspect(context, value).getByteList());
+                        str.append(',').append(' ');
+                    }
+                }
+
+                str = new ByteList(str, 0, str.length() - 2); /* drop the last ", " */
+                str.append(')');
+            }
+        }
+
+        str.append(')');
+
+        return RubyString.newStringLight(context.runtime, str);
```

Comment:
Probably needs a specific encoding? `new ByteList` defaults to `ASCII-8BIT` and I believe `inspect` is supposed to always be `US-ASCII`. If there's any characters not in that character range they'd need to be encoded.

Reply:
I have confirmed same behavior as ruby 2.6.3. :confused:❓ 

```
% ruby -v
ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-darwin19]
2.6.3 :001 > (1..).step.inspect.encoding
 => #<Encoding:ASCII-8BIT>
```

Reply:
Ok, I stand corrected!